lib/customer-deposit-wallet/src/Cardano/Wallet/Deposit/Pure/State/Payment.hs

```diff
@@ -55,15 +78,108 @@ import qualified Data.Text as T
 
 data ErrCreatePayment
     = ErrCreatePaymentNotRecentEra (Read.EraValue Read.Era)
-    | ErrCreatePaymentBalanceTx (Write.ErrBalanceTx Write.Conway)
+    | ErrNotEnoughAda { shortfall :: Value }
+    | ErrEmptyUTxO
+
+    | ErrTxOutAdaInsufficient { outputIx :: Int, suggestedMinimum :: Coin }
+
+    -- | The final balanced tx was too big. Either because the payload was too
+    -- big to begin with, or because we failed to select enough inputs without
+    -- making it too big, e.g. due to the UTxO containing lots of dust.
+    --
+    -- We should ideally split out 'TooManyPayments' from this error.
+    -- We should ideally also be able to create payments even when dust causes
+    -- us to need preparatory txs.
+    | ErrTxMaxSizeLimitExceeded
     deriving (Eq, Show)
 
+translateBalanceTxError :: Write.ErrBalanceTx Write.Conway -> ErrCreatePayment
+translateBalanceTxError = \case
+    Write.ErrBalanceTxAssetsInsufficient
+        Write.ErrBalanceTxAssetsInsufficientError{shortfall} ->
+            ErrNotEnoughAda
+                { shortfall = fromLedgerValue shortfall
+                }
+    Write.ErrBalanceTxMaxSizeLimitExceeded ->
+            ErrTxMaxSizeLimitExceeded
+    Write.ErrBalanceTxExistingKeyWitnesses _ ->
+        impossible "ErrBalanceTxExistingKeyWitnesses"
+    Write.ErrBalanceTxExistingCollateral ->
+        impossible "ErrBalanceTxExistingCollateral"
+    Write.ErrBalanceTxExistingTotalCollateral ->
+        impossible "ErrBalanceTxExistingTotalCollateral"
+    Write.ErrBalanceTxExistingReturnCollateral ->
+        impossible "ErrBalanceTxExistingReturnCollateral"
+    Write.ErrBalanceTxInsufficientCollateral _ ->
+        impossible "ErrBalanceTxInsufficientCollateral"
+    Write.ErrBalanceTxAssignRedeemers _ ->
+        impossible "ErrBalanceTxAssignRedeemers"
+    Write.ErrBalanceTxInternalError e ->
+        impossible $ show e
+    Write.ErrBalanceTxInputResolutionConflicts _ ->
+        -- We are never creating partialTxs with pre-selected inputs, which
+        -- means this is impossible.
+        impossible "conflicting input resolution"
+    Write.ErrBalanceTxUnresolvedInputs _ ->
+        -- We are never creating partialTxs with pre-selected inputs, which
+        -- means this is impossible.
+        impossible "unresolved inputs"
+    Write.ErrBalanceTxUnresolvedRefunds _ ->
+        impossible "unresolved refunds"
+    Write.ErrBalanceTxOutputError (Write.ErrBalanceTxOutputErrorOf ix (Write.ErrBalanceTxOutputAdaQuantityInsufficient{minimumExpectedCoin})) ->
+        ErrTxOutAdaInsufficient { outputIx = ix, suggestedMinimum = minimumExpectedCoin }
+    Write.ErrBalanceTxOutputError (Write.ErrBalanceTxOutputErrorOf _ix (Write.ErrBalanceTxOutputSizeExceedsLimit{})) ->
+        impossible "value can't be too big if there are no assets"
```

Comment:
Hm, but `createPayment` does actually take Values and not Coins... This is possible then...